automatic/keeweb/update.ps1

```diff
@@ -22,48 +20,36 @@ function global:au_SearchReplace {
 
 function global:au_GetLatest {
 
-    $downloadedPage = Invoke-WebRequest -Uri $releases -UseBasicParsing
-
-    $baseUrl = $([System.Uri]$releases).Authority
-    $scheme = $([System.Uri]$releases).Scheme
-
-    $url32 = $downloadedPage.links | ? href -match '32.exe$' | select -First 1 -expand href
-    if ($url32.Authority -cnotmatch $baseUrl) {
-        $url32 = $scheme + '://' + $baseUrl + $url32
-    }
-    $url32SegmentSize = $([System.Uri]$url32).Segments.Length
-    $filename32 = $([System.Uri]$url32).Segments[$url32SegmentSize - 1]
-
-    $url64 = $downloadedPage.links | ? href -match '64.exe$' | select -First 1 -expand href
-    if ($url64.Authority -cnotmatch $baseUrl) {
-        $url64 = $scheme + '://' + $baseUrl + $url64
+    $jsonAnswer = (Invoke-WebRequest -Uri "https://api.github.com/repos/keeweb/keeweb/releases/latest" -UseBasicParsing).Content | ConvertFrom-Json
+    $version = $jsonAnswer.tag_name -Replace '[^0-9.]'
+    $jsonAnswer.assets |Where { $_.name -Match "(32|x64).exe" } | ForEach-Object {
+        if ($_.browser_download_url -Match 'x64') {
+            $url64 = $_.browser_download_url
+            $filename64 = $_.name
+        } else {
+            $url32 = $_.browser_download_url
+            $filename32 = $_.name
+        }
     }
-    $url64SegmentSize = $([System.Uri]$url64).Segments.Length
-    $filename64 = $([System.Uri]$url64).Segments[$url64SegmentSize - 1]
-
-    $version = [regex]::match($url32,'/[A-Za-z]+-([0-9]+.[0-9]+.[0-9]+).*exe').Groups[1].Value
 
     # Since upstream is providing hashes, let's rely on these instead.
     # i.e. this adds some added value as we are making sure the hashes are
     # really those computed by upstream. We are not simply relying on the hashs
     # provided by the cmdlet Get-RemoteFiles.
-    $verifyUrl = $releases + '/download/VERSION/Verify.sha256'
-    $verifyUrl = $verifyUrl -Replace 'VERSION',"v$version"
+    $verifyUrl = ($jsonAnswer.assets | Where { $_.name -Match "Verify.sha256" }).browser_download_url
     $downloadedPage = Invoke-WebRequest -Uri $verifyUrl -UseBasicParsing
     $downloadedPage = [System.Text.Encoding]::UTF8.GetString($downloadedPage.Content)
     $downloadedPage = $downloadedPage.Split([Environment]::NewLine)
-
     foreach ($line in $downloadedPage) {
-
-        $parsed = $line -split ' |\n' -replace '\*',''
-
-        if ($parsed[1] -cmatch $filename32) {
+        $parsed = $line -Split ' |\n' -Replace '\*',''
+        if ($parsed[1] -Match $filename32) {
             $hash32 = $parsed[0]
         }
-        if ($parsed[1] -cmatch $filename64) {
+        if ($parsed[1] -Match $filename64) {
```

Comment:
`cmatch` is used on purpose here, as that is case sensitive match. By default, PowerShell uses case insensitive matching.

Reply:
Taken care of.